compiler/ast/ast.go

```diff
@@ -579,13 +579,17 @@ type (
 		Loc  `json:"loc"`
 	}
 	Join struct {
-		Kind       string      `json:"kind" unpack:""`
-		Style      string      `json:"style"`
-		RightInput Seq         `json:"right_input"`
-		Cond       JoinExpr    `json:"cond"`
-		Args       Assignments `json:"args"`
+		Kind       string     `json:"kind" unpack:""`
+		Style      string     `json:"style"`
+		Alias      *JoinAlias `json:"alias"`
+		RightInput Seq        `json:"right_input"`
+		Cond       JoinExpr   `json:"cond"`
```

Comment:
Nit: Keep the order here (and elsewhere) in sync with PEG rule?

```suggestion
		Style      string     `json:"style"`
		RightInput Seq        `json:"right_input"`
		Alias      *JoinAlias `json:"alias"`
		Cond       JoinExpr   `json:"cond"`
```